core/src/test/java/com/radixdlt/p2p/addressbook/AddressBookMigrationTest.java

```diff
@@ -0,0 +1,164 @@
+import com.radixdlt.monitoring.Metrics;
+import com.radixdlt.monitoring.MetricsInitializer;
+import com.radixdlt.rev2.NodeRustEnvironmentBuilder;
+import com.radixdlt.rev2.modules.AddressBookModule;
+import com.radixdlt.serialization.Serialization;
+import com.radixdlt.serialization.core.ClasspathScanningSerializationPolicy;
+import com.radixdlt.serialization.core.ClasspathScanningSerializerIds;
+import com.radixdlt.store.BerkeleyDbDefaults;
+import com.radixdlt.store.NodeStorageLocation;
+import com.radixdlt.utils.properties.RuntimeProperties;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import org.apache.commons.cli.ParseException;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public class AddressBookMigrationTest {
+  @Rule public TemporaryFolder folder = new TemporaryFolder();
+
+  private Module createCommonModule() {
+    return new AbstractModule() {
+      @Provides
+      @Singleton
+      @NodeStorageLocation
+      String nodeStorageLocation() {
+        return folder.getRoot().getPath();
+      }
+
+      @Provides
+      @Singleton
+      NodeRustEnvironment nodeRustEnvironment() throws IOException {
+        return NodeRustEnvironmentBuilder.createNodeRustEnvironment(folder.newFolder().getPath());
+      }
+
+      @Provides
+      @Singleton
+      Metrics metrics() {
+        return new MetricsInitializer().initialize();
+      }
+
+      @Provides
+      @Singleton
+      Serialization serialization() {
+        return Serialization.create(
+            ClasspathScanningSerializerIds.create(), ClasspathScanningSerializationPolicy.create());
+      }
+
+      @Provides
+      @Singleton
+      RuntimeProperties properties() throws ParseException {
+        return RuntimeProperties.defaultWithOverrides(Map.of());
+      }
+    };
+  }
+
+  private Module createBerkeleyModule() {
+    return new AbstractModule() {
+      @Provides
+      @Singleton
+      BerkeleyAddressBookStore addressBookPersistence(
+          RuntimeProperties properties,
+          Serialization serialization,
+          Metrics metrics,
+          @NodeStorageLocation String nodeStorageLocation) {
+        return new BerkeleyAddressBookStore(
+            serialization,
+            metrics,
+            nodeStorageLocation,
+            BerkeleyDbDefaults.createDefaultEnvConfigFromProperties(properties));
+      }
+    };
+  }
+
+  @Test
+  public void address_book_migrates_successfully() {
+    var injector =
+        Guice.createInjector(createCommonModule(), new AddressBookModule(), createBerkeleyModule());
+    var addresses = Set.of(newAddressBookEntry(1), newAddressBookEntry(2), newAddressBookEntry(3));
+
+    // Create some entries in the Berkeley address book
+    try (var berkeleyAddressBook = injector.getInstance(BerkeleyAddressBookStore.class)) {
+      addresses.forEach(berkeleyAddressBook::upsertEntry);
```

Comment:
Nice test - but I notice it's not complete? Can we also add some high priority peers?

Reply:
Fixed